src/auto_scheduler/search_policy/sketch_policy_rules.cc

```diff
@@ -908,7 +795,362 @@ InitPopulationRule::ResultKind InitThreadBind::Apply(SketchPolicyNode* policy, S
+        if (!target_is_tiled) {  // Do not go into reduce iter
+          break;
+        }
+      } else if (target_iter->iter_kind == IteratorKind::kSpatial) {
+        if (visited_reduce) {  // Do not go into inner tile
+          break;
+        }
+      }
+
+      if (target_iter->annotation == IteratorAnnotation::kUnroll) {
+        // Do not go into the unroll region of const tensor indices
+        break;
+      }
+
+      if (GetExtent(target_iter) == 1) {
+        // Skip iterators with length of 1
+        continue;
+      }
+      if (target_compute_at_other && target_iter->iter_kind == IteratorKind::kSpatial &&
+          StrEndsWith(target_iter->name, ".0")) {
+        // Skip the first level iterators if target stage compute_at another stage
+        // In this case, the lengths of first level iterators are always one
+        continue;
+      }
+      candidates.emplace_back(target_stage_id, i);
+
+      if ((*state)->attach_map->iter_to_attached_stages.count(std::make_pair(target_stage_id, i))) {
+        break;
+      }
+    }
+
+    // if the target_stage is already compute_at another stage X, try also compute_at X
+    // We call stage X as `target_target_stage`
+    if (target_compute_at_other) {
+      int target_target_stage_id;
+      target_target_stage_id = (*state)->attach_map->stage_to_attach_iter.at(target_stage_id).first;
+      const Stage& target_target_stage = (*state)->stages[target_target_stage_id];
+
+      for (size_t i = 0; i < target_target_stage->iters.size(); ++i) {
+        const Iterator& target_target_iter = target_target_stage->iters[i];
+        if (target_target_iter->iter_kind == IteratorKind::kReduction ||
+            (*state)->attach_map->iter_to_attached_stages.count(
+                std::make_pair(target_target_stage_id, i))) {
+          break;
+        }
+
+        if (target_target_iter->annotation == IteratorAnnotation::kUnroll) {
+          // Do not go into the unroll region of const tensor indices
+          break;
+        }
+
+        if (GetExtent(target_target_iter) == 1) {  // skip iterators with length of 1
+          continue;
+        }
+
+        candidates.emplace_back(target_target_stage_id, i);
+      }
+    }
+
+    int choice = (policy->rand_gen)() % (candidates.size() + 2);
+
+    if (choice == 0) {
+      if (!HasReduceIter(stage)) {
+        const auto& stage_to_attach_iter = (*state)->attach_map->stage_to_attach_iter;
+        if (stage_to_attach_iter.find(stage_id) != stage_to_attach_iter.end()) {
+          state->compute_inline(stage_id);
+        }
+      }
+    } else if (choice == 1) {
+      state->compute_root(stage_id);
+    } else {
+      choice = choice - 2;
+      const Stage& stage = (*state)->stages[candidates[choice].first];
+      state->compute_at(stage_id, candidates[choice].first,
+                        stage->iters[candidates[choice].second]);
+    }
+  }
+
+  *state = policy->search_task->compute_dag.InferBound(*state);
```

Comment:
This InferBound is required during SampleInitPopulation. However, it introduces a redundant InferBound during EvolutionarySearch, we should remove it during EvolutionarySearch.

Reply:
Good catch. Miss this line when moving the function.

Reply:
Hmm I realized that it's hard to recognize if this is being used by initial population or mutation. Do you have any suggestions? Or should we just simply remove this mutation in the initial population as @jcf94 suggested?

Reply:
@comaniac  We should not remove it.
To implement this, we can create a new function for the common part. We then create two new rules and use them separately.